Parse/src/main/java/com/parse/ParseOkHttpClient.java

```diff
@@ -48,17 +47,18 @@
 
   public ParseOkHttpClient(int socketOperationTimeout, SSLSessionCache sslSessionCache) {
 
-    okHttpClient = new OkHttpClient();
+    OkHttpClient.Builder builder = new OkHttpClient.Builder();
 
-    okHttpClient.setConnectTimeout(socketOperationTimeout, TimeUnit.MILLISECONDS);
-    okHttpClient.setReadTimeout(socketOperationTimeout, TimeUnit.MILLISECONDS);
+    builder.connectTimeout(socketOperationTimeout, TimeUnit.MILLISECONDS);
+    builder.readTimeout(socketOperationTimeout, TimeUnit.MILLISECONDS);
 
     // Don't handle redirects. We copy the setting from AndroidHttpClient.
     // For detail, check https://quip.com/Px8jAxnaun2r
-    okHttpClient.setFollowRedirects(false);
+    builder.followRedirects(false);
 
-    okHttpClient.setSslSocketFactory(SSLCertificateSocketFactory.getDefault(
-        socketOperationTimeout, sslSessionCache));
+    // Broken until OkHttp3 can expose trust managers -- https://github.com/square/okhttp/commit/85f74e2004eaf0d4ff339e8644df3d8e716361e5
+    // builder.sslSocketFactory(SSLCertificateSocketFactory.getDefault(socketOperationTimeout, sslSessionCache));
```

Comment:
I don't have enough experience to know whether SSL caching functionality is a blocker... anyone on the Parse team could provide some insight?